HHS/Head-Start-TTADP
# [TTAHUB-3640] Allow setting goal source when it is null, even when used on approved AR

## Description of change

After  talking with OHS about this bug, we all agreed to allow setting a goal source, even if the goal has been used on an approved AR, in the case that it has a null source. Without this change, the goal source isn't settable anywhere, and the AR cannot be saved because source is null.

## How to test

With prod data, impersonate the user and view the AR specified in 3640. With this PR, you should be able to set the goal source (even though the goal is used on an approved AR) and save the report.

You should still not be able to set a goal source on a multi-recipient AR.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3640


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
